scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java

```diff
@@ -99,12 +99,12 @@ public static LauncherProperties fromCommandLineArguments(String[] args) {
 
     Map<ParamName, LauncherProperty> properties = new EnumMap<>(ParamName.class);
     for (String arg : args) {
-      String[] propertyNameValuePair = arg.split(PROPERTY_NAME_VALUE_DELIMITER);
+      String[] propertyNameValuePair = arg.split(PROPERTY_NAME_VALUE_DELIMITER, 2);
       ParamName propertyName = ParamName.fromPropertyName(propertyNameValuePair[0]);
       String propertyValue = propertyNameValuePair[1];
 
       if(propertyName == null) {
-        continue;
+        throw new LauncherInitializationException("Unsupported property");
```

Comment:
I think that this check should be relocated to `isArgsInValidFormat`. Also, property name can be non-null but still unsupported

Reply:
ParamName.fromPropertyName(propertyNameValuePair[0]) returns null if name is unsupported

Reply:
Ok, I agree

Reply:
If I relocate it I will be calling ParamName.fromPropertyName 2 times for every property. The format is valid, but the properties are unsupported. It returns null when unsupported.

Reply:
It's not a big deal. You also, already split it twice.